File: code/client/web/src/Domain/wrappers/identifier/IdentifierValidator.ts
import { IdentifierValidationError } from './IdentifierValidationError';

export class IdentifierValidator {
    validate(value: number): IdentifierValidationError[] {
        const errors: IdentifierValidationError[] = [];

        if (value < 0) {
            errors.push(IdentifierValidationError.NegativeValue);
        }

        if (errors.length > 0) {
            return errors;
        }

        return [];
    }
}

File: code/server/jvm/im-domain/src/main/kotlin/im/domain/wrappers/identifier/IdentifierValidator.kt
package im.domain.wrappers.identifier

import im.domain.Either
import im.domain.failure
import im.domain.success

class IdentifierValidator {
    fun validate(value: Long): Either<List<IdentifierValidationError>, Unit> {
        val errors = mutableListOf<IdentifierValidationError>()

        if (value < 0) {
            errors.add(IdentifierValidationError.NegativeValue)
        }

        if (errors.isNotEmpty()) {
            return failure(errors)
        }

        return success(Unit)
    }
}
